django/contrib/admin/options.py

```diff
@@ -1058,7 +1058,14 @@ def response_add(self, request, obj, post_url_continue=None):
         opts = obj._meta
         pk_value = obj._get_pk_val()
         preserved_filters = self.get_preserved_filters(request)
-        msg_dict = {'name': force_text(opts.verbose_name), 'obj': force_text(obj)}
+        obj_url = reverse('admin:%s_%s_change' %
+                          (opts.app_label, opts.model_name),
+                          args=(quote(pk_value),),
+                          current_app=self.admin_site.name)
+        msg_dict = {
+            'name': force_text(opts.verbose_name),
+            'obj': force_text('<a href="{0}">{1}</a>').format(urlquote(obj_url), escape(obj))
```

Comment:
@kezabelle 
Indeed, when "Save and continue" is used to save object, link in post save message will be pointing to the very same page user is already on.

I think that it is better to leave it this way to not introduce any mess into "Post save message contains link to object's edit form" assertion. 
Besides that, this links does not do any harm (changed `color` and `cursor` css properties for object name does not look so scary) and even can be useful sometimes, like for copying the link on mobile devices, where address bar can often be hidden.